pybop/models/base_model.py

```diff
@@ -742,34 +736,18 @@ def submesh_types(self):
     def mesh(self):
         return self._mesh
 
-    @mesh.setter
-    def mesh(self, mesh):
-        self._mesh = mesh if mesh is not None else None
-
     @property
     def disc(self):
         return self._disc
 
-    @disc.setter
-    def disc(self, disc):
-        self._disc = disc if disc is not None else None
-
     @property
     def var_pts(self):
         return self._var_pts
 
-    @var_pts.setter
-    def var_pts(self, var_pts: Optional[dict[str, int]]):
-        self._var_pts = var_pts.copy() if var_pts is not None else None
-
     @property
     def spatial_methods(self):
         return self._spatial_methods
 
     @property
     def solver(self):
         return self._solver
-
-    @solver.setter
```

Comment:
Let's keep the solver setter, as it's helpful for an upcoming IDAKLU example I have on a local branch.

Reply:
Thanks for the suggestion, I'm going to continue as I have already prepared #425 to merge next but please feel free to add it back into your local branch.